An error occurred while executing the following cell:
------------------
import numpy as np
import matplotlib.pyplot as plt
import os

try:
    import tensorflow
except ModuleNotFoundError:
    %pip install -qq tensorflow
    import tensorflow
import tensorflow as tf
from tensorflow import keras

assert tf.__version__ >= "2.0"


K = tf.keras.backend
------------------

---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/tmp/ipykernel_6446/3040475755.py in <module>
      5 try:
----> 6     import tensorflow
      7 except ModuleNotFoundError:

ModuleNotFoundError: No module named 'tensorflow'

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_6446/3040475755.py in <module>
      7 except ModuleNotFoundError:
      8     get_ipython().run_line_magic('pip', 'install -qq tensorflow')
----> 9     import tensorflow
     10 import tensorflow as tf
     11 from tensorflow import keras

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/__init__.py in <module>
     49 from ._api.v2 import autograph
     50 from ._api.v2 import bitwise
---> 51 from ._api.v2 import compat
     52 from ._api.v2 import config
     53 from ._api.v2 import data

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/__init__.py in <module>
     35 import sys as _sys
     36 
---> 37 from . import v1
     38 from . import v2
     39 from tensorflow.python.compat.compat import forward_compatibility_horizon

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/__init__.py in <module>
     28 from . import autograph
     29 from . import bitwise
---> 30 from . import compat
     31 from . import config
     32 from . import data

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/compat/__init__.py in <module>
     35 import sys as _sys
     36 
---> 37 from . import v1
     38 from . import v2
     39 from tensorflow.python.compat.compat import forward_compatibility_horizon

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/compat/v1/__init__.py in <module>
     45 from tensorflow._api.v2.compat.v1 import layers
     46 from tensorflow._api.v2.compat.v1 import linalg
---> 47 from tensorflow._api.v2.compat.v1 import lite
     48 from tensorflow._api.v2.compat.v1 import logging
     49 from tensorflow._api.v2.compat.v1 import lookup

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/lite/__init__.py in <module>
      7 
      8 from . import constants
----> 9 from . import experimental
     10 from tensorflow.lite.python.lite import Interpreter
     11 from tensorflow.lite.python.lite import OpHint

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/lite/experimental/__init__.py in <module>
      6 import sys as _sys
      7 
----> 8 from . import authoring
      9 from tensorflow.lite.python.analyzer import ModelAnalyzer as Analyzer
     10 from tensorflow.lite.python.lite import OpResolverType

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/_api/v2/compat/v1/lite/experimental/authoring/__init__.py in <module>
      6 import sys as _sys
      7 
----> 8 from tensorflow.lite.python.authoring.authoring import compatible

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/lite/python/authoring/authoring.py in <module>
     41 
     42 # pylint: disable=g-import-not-at-top
---> 43 from tensorflow.lite.python import convert
     44 from tensorflow.lite.python import lite
     45 from tensorflow.lite.python.metrics import converter_error_data_pb2

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/lite/python/convert.py in <module>
     25 
     26 from tensorflow.lite.python import lite_constants
---> 27 from tensorflow.lite.python import util
     28 from tensorflow.lite.python import wrap_toco
     29 from tensorflow.lite.python.convert_phase import Component

~/miniconda3/envs/py37/lib/python3.7/site-packages/tensorflow/lite/python/util.py in <module>
     51 # pylint: disable=unused-import
     52 try:
---> 53   from jax import xla_computation as _xla_computation
     54 except ImportError:
     55   _xla_computation = None

~/miniconda3/envs/py37/lib/python3.7/site-packages/jax/__init__.py in <module>
     33 # We want the exported object to be the class, so we first import the module
     34 # to make sure a later import doesn't overwrite the class.
---> 35 from jax import config as _config_module
     36 del _config_module
     37 

~/miniconda3/envs/py37/lib/python3.7/site-packages/jax/config.py in <module>
     15 # TODO(phawkins): fix users of this alias and delete this file.
     16 
---> 17 from jax._src.config import config

~/miniconda3/envs/py37/lib/python3.7/site-packages/jax/_src/config.py in <module>
     27 from absl import logging
     28 
---> 29 from jax._src import lib
     30 from jax._src.lib import jax_jit
     31 from jax._src.lib import transfer_guard_lib

~/miniconda3/envs/py37/lib/python3.7/site-packages/jax/_src/lib/__init__.py in <module>
     95   jax_version=jax.version.__version__,
     96   jaxlib_version=jaxlib.version.__version__,
---> 97   minimum_jaxlib_version=jax.version._minimum_jaxlib_version)
     98 
     99 

~/miniconda3/envs/py37/lib/python3.7/site-packages/jax/_src/lib/__init__.py in check_jaxlib_version(jax_version, jaxlib_version, minimum_jaxlib_version)
     87            f'incompatible with jax version {jax_version}. Please '
     88            'update your jax and/or jaxlib packages.')
---> 89     raise RuntimeError(msg)
     90 
     91   return _jaxlib_version

RuntimeError: jaxlib version 0.3.25 is newer than and incompatible with jax version 0.3.14. Please update your jax and/or jaxlib packages.
RuntimeError: jaxlib version 0.3.25 is newer than and incompatible with jax version 0.3.14. Please update your jax and/or jaxlib packages.
